src/Elastic.CommonSchema.LoggerProvider/ElasticsearchDataProcessor.cs

```diff
@@ -0,0 +1,262 @@
+				var type = entryAssemblyName.Name;
+				var versionAttribute = entryAssembly.GetCustomAttributes(false)
+					.OfType<AssemblyInformationalVersionAttribute>()
+					.FirstOrDefault();
+				var version = versionAttribute?.InformationalVersion ?? entryAssemblyName.Version.ToString();
+				_service = new Service { Type = type, Version = version };
+			}
+
+			return _service;
+		}
+
+		internal int GetSeverity(LogLevel logLevel)
+		{
+			switch (logLevel)
+			{
+				case LogLevel.Critical:
+					return 2;
+				case LogLevel.Error:
+					return 3;
+				case LogLevel.Warning:
+					return 4;
+				case LogLevel.Information:
+					return 6;
+				case LogLevel.Debug:
+				case LogLevel.Trace:
+					return 7;
+				default:
+					return 7;
+			}
+		}
+
+		private void PostEvent(LogEvent logEvent)
+		{
+			var indexTime = logEvent.Timestamp ?? ElasticsearchLoggerProvider.LocalDateTimeProvider();
+			if (_options.IndexOffset.HasValue) indexTime = indexTime.ToOffset(_options.IndexOffset.Value);
+
+			var index = string.Format(_options.Index, indexTime);
+
+			var id = Guid.NewGuid().ToString();
+
+			var localClient = _lowLevelClient;
+			var response = localClient.Index<StringResponse>(index, id, PostData.Serializable(logEvent));
+		}
+
+		private void ProcessLogQueue()
+		{
+			try
+			{
+				foreach (var logEvent in _messageQueue.GetConsumingEnumerable()) PostEvent(logEvent);
+			}
+			catch
+			{
+				try
+				{
+					_messageQueue.CompleteAdding();
+				}
+				catch { }
+			}
+		}
+
+		private void UpdateClient()
+		{
+			// TODO: Check if Uri has changed before recreating
+			// TODO: Injectable factory? Or some way of testing.
+
+			ConnectionConfiguration settings;
+			if (_options.NodeUris.Length == 0)
+			{
+				// This is SingleNode with "http://localhost:9200"
+				settings = new ConnectionConfiguration();
+			}
+			else if (_options.ConnectionPoolType == ConnectionPoolType.SingleNode
+				|| _options.ConnectionPoolType == ConnectionPoolType.Unknown && _options.NodeUris.Length == 1)
+				settings = new ConnectionConfiguration(_options.NodeUris[0]);
+			else
+			{
+				IConnectionPool connectionPool;
+				switch (_options.ConnectionPoolType)
+				{
+					// TODO: Add option to randomize pool
+					case ConnectionPoolType.Unknown:
+					case ConnectionPoolType.Sniffing:
+						connectionPool = new SniffingConnectionPool(_options.NodeUris);
+						break;
+					case ConnectionPoolType.Static:
+						connectionPool = new StaticConnectionPool(_options.NodeUris);
+						break;
+					case ConnectionPoolType.Sticky:
+						connectionPool = new StickyConnectionPool(_options.NodeUris);
+						break;
+					// case ConnectionPoolType.StickySniffing:
+					// case ConnectionPoolType.Cloud:
+					default:
+						throw new NotSupportedException($"Unknown connection pool type {_options.ConnectionPoolType}");
+				}
+
+				settings = new ConnectionConfiguration(connectionPool);
+			}
+
+			var lowlevelClient = new ElasticLowLevelClient(settings);
```

Comment:
```suggestion
			var lowlevelClient = new ElasticLowLevelClient(settings);
			((ElasticLowLevelClient)_lowLevelClient).Settings.Dispose();
```

Some background: https://github.com/elastic/elasticsearch-net/issues/4018

We initially did not want to advertise `IDisposable` on the client since we want folks to reuse instances agressively. We will make the client disposable with `8.0.0` most likely.

Reply:
Reloading the client will affect in flight requests and could potentially throw abort exceptions. 

Is this something we really need to support or adding in because the reload trigger is there? 

We could potentially only support reloading new endpoints with:

```
((ElasticLowLevelClient)_lowLevelClient).Settings.ConnectionPool.Reseed(nodes);
```

Without destroying the client, other settings would require a restart though.

Reply:
Not sure I understand your suggestion. If you are saying I should dispose of the client (or it's child), then I would do that after the atomic/interlocked swap, i.e. get rid of the old one.

PostEvent does grab a local copy, and configuration would not reload very often (if ever), so leaving one old one around for GC Disposable shouldn't be an issue. 

And, as you said, if you dispose of it while there is an inflight request that might cause an issue (which is why I create a new one, then do an atomic swap).

Reply:
If only nodes changed, then it would be good to just reload them -- a good future enhancement.

Reply:
If you have disposable fields, then the client should probably be disposable, even if intended to be a singleton. But you can also have a explicit implementation of the interface which makes it slightly less likely users will just be disposing you.

And I'm aware of the damned-if-you-do-damned-if-you-dont issues with either disposing or not disposing of HttpClientHandler... the improvements in 3.0/3.1 seem to handle it quite well with the single HttpClient / internal pool of connections, and having middleware for things like test injection, policy based retries, etc.

Reply:
I suggest also move this to an issue, for a future change. (Although I'm not sure there is a better solution, especially given config changes would be rare.)